lib/kb_ObjectInfo/kb_ObjectInfoImpl.py

```diff
@@ -518,21 +541,71 @@ def genomeset_report(self, ctx, params):
         multi_fasta = []
         rpt_string = ''
         
+#       Always do the list of genomes - short and ensures that there is an HTML link
+        gsr = CreateMultiGenomeReport(self.config)
+        rpt_list = gsr.readGenomeSet(genome_name, genomeset_data)
+        htmltable = self.make_HTML(rpt_list,'col_header')
+        html_report_path = os.path.join(self.scratch, 'genomeset_comparison.html')
+        html_report_txt = open(html_report_path, "w")
+        html_report_txt.write("<h1>GENOME COMPARISON</h1>")
+        html_report_txt.write(htmltable)
+        html_report_txt.close()
+        
         if params['showGenomes']:
-            gsr = CreateMultiGenomeReport(self.config)
-            rpt_list = gsr.readGenomeSet(genome_name, genomeset_data)
             rpt_string += self.write_to_file(rpt_list,'genomeset_tab_file.tsv',"\t")
             self.write_to_file(rpt_list,'genomeset_cvs_file.csv',",")
-            htmltable = self.make_HTML(rpt_list,'col_header')
-            
-            html_report_path = os.path.join(self.scratch, 'genomeset_comparison.html')
+
+        if params['listCoding']:
+            cf = CreateFeatureLists(self.config)
+            myGS = genomeset_data['elements']
+            rpt_list = []
+            for ele in myGS:
+                genome = self.dfu.get_objects({'object_refs': [myGS[ele]['ref']]})
+                genome_data = genome['data'][0]['data']
+
+                rpt_list += cf.delimitedTable(genome_data, 'features')
```

Comment:
Again, this would have been good advise back when the module was first ported over to kbaseapps. This takes me back to learning COBOL but it does make a good point. Perhaps I took the shorthand a little too far this time. I've done what I could see in the time available.